scripts/autotools/libmongoc/CheckSSL.m4

```diff
@@ -1,23 +1,31 @@
 PHP_ARG_WITH([mongodb-ssl],
              [whether to enable crypto and TLS],
-             [AS_HELP_STRING([--with-mongodb-ssl=@<:@auto/no/openssl/libressl/darwin@:>@],
+             [AS_HELP_STRING([--with-mongodb-ssl=@<:@auto/openssl/libressl/darwin/no@:>@],
                              [MongoDB: Enable TLS connections and SCRAM-SHA-1 authentication [default=auto]])],
              [auto],
              [no])
 
+if test "$PHP_MONGODB_SSL" = "auto" -o "$PHP_MONGODB_SSL" = "no"; then
+  crypto_required="no"
+else
+  crypto_required="yes"
+
+  dnl PHP_ARG_WITH without a value assigns "yes". Treat it like "auto" but
+  dnl require a crypto library.
+  if test "$PHP_MONGODB_SSL" = "yes"; then
+    PHP_MONGODB_SSL="auto"
+  fi
+fi
+
+PHP_MONGODB_VALIDATE_ARG([PHP_MONGODB_SSL], [auto openssl libressl darwin no])
+
 PHP_ARG_WITH([openssl-dir],
```

Comment:
I still intend to remove this in [PHPC-2309](https://jira.mongodb.org/browse/PHPC-2309), and that is the reason I did not bother to add validation. Also, that'd be pretty hard to do given that it accepts an arbitrary string for a directory.